react-components/src/components/Architecture/AnchoredDialog.tsx

```diff
@@ -16,21 +16,35 @@ export const AnchoredDialog = (): ReactNode => {
   AnchoredDialogUpdater.setCounterDelegate(setActiveToolUpdater);
 
   const renderTarget = useRenderTarget();
-  if (renderTarget === undefined) {
-    return <></>;
-  }
   const activeTool = renderTarget.commandsController.activeTool;
-  if (activeTool === undefined) {
-    return <></>;
-  }
-
-  const dialogContent = useMemo(() => activeTool.getAnchoredDialogContent(), [activeToolUpdate]);
+  const dialogContent = useMemo(() => activeTool?.getAnchoredDialogContent(), [activeToolUpdate]);
   const isSomeEnabled = dialogContent?.contentCommands.some((command) => command.isEnabled);
 
+  useEffect(() => {
+    if (dialogContent?.customListeners === undefined) return;
+    dialogContent.customListeners.forEach((listener) => {
+      window.addEventListener(listener.eventName, listener.callback, true);
+    });
+    return () => {
+      if (dialogContent?.customListeners === undefined) return;
+      dialogContent.customListeners.forEach((listener) => {
+        window.removeEventListener(listener.eventName, listener.callback, true);
+      });
+    };
+  }, [dialogContent]);
+
   if (dialogContent === undefined || isSomeEnabled !== true) {
     return undefined;
   }
 
+  if (renderTarget === undefined) {
+    return <></>;
+  }
+
+  if (activeTool === undefined) {
+    return <></>;
+  }
```

Comment:
I think we can just combine all of these to one if-check? :thinking:

Reply:
yep
https://github.com/cognitedata/reveal/pull/4909/commits/07d25883305ba05951442a30abf05d9f9ffe1757